packages/peregrine/lib/talons/Logo/useLogo.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+
+/**
+ *
+ * @param {*} props.query the footer data query
+ */
+export const useLogo = props => {
+    const { query, defaultSrc, defaultWidth, defaultHeight, defaultAlt } = props;
+    const { error, data } = useQuery(query);
+
+    useEffect(() => {
+        if (error) {
+            console.log('Error fetching logo data.');
+        }
+    }, [error]);
+
+    let logoSrc;
+
+    if (data && data.storeConfig && data.storeConfig.header_logo_src) {
+        logoSrc =
+            (data &&
+                data.storeConfig &&
+                data.storeConfig.secure_base_media_url) +
+            'logo/' +
+            (data && data.storeConfig && data.storeConfig.header_logo_src);
+    } else {
+        logoSrc = defaultSrc;
+    }
+
+    return {
+        configSrc: logoSrc,
```

Comment:
I'd probably also rename these and remove `config` ie from `configSrc` to just `src`.

Reply:
Suggestion added.